xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -15140,14 +15140,16 @@ glusterd_check_brick_order(dict_t *dict, char *err_str, int32_t type,
 out:
     ai_list_tmp2 = NULL;
     GF_FREE(brick_list_ptr);
-    cds_list_for_each_entry(pre_list_tmp1, &pre_list->list, list)
-    {
-        if (pre_list_tmp1->info)
-            freeaddrinfo(pre_list_tmp1->info);
-        free(ai_list_tmp2);
-        ai_list_tmp2 = pre_list_tmp1;
+    if (pre_list != NULL) {
+        cds_list_for_each_entry(pre_list_tmp1, &pre_list->list, list)
+        {
+            if (pre_list_tmp1->info)
+                freeaddrinfo(pre_list_tmp1->info);
+            free(ai_list_tmp2);
+            ai_list_tmp2 = pre_list_tmp1;
```

Comment:
While reviewing other patch, I realised that ai_list_tmp2  is not being freed here once the loop ends.

Reply:
@aspandey It's being freed at line 15161

Reply:
same comment as I gave for other patch...
we should use ai_list_tmp2 for this "if" and free it here only and then use it in next "if" at  15153 and free it there.